toolkit/tools/pkg/imagecustomizerlib/cosicommon.go

```diff
@@ -303,3 +323,39 @@ func getArchitectureForCosi() string {
 	}
 	return runtime.GOARCH
 }
+
+func getAllPackagesFromChroot(imageConnection *ImageConnection) ([]OsPackage, error) {
+	var packages []OsPackage
+
+	err := imageConnection.Chroot().UnsafeRun(func() error {
+		out, _, err := shell.Execute(
+			"rpm", "-qa", "--queryformat", "%{NAME} %{VERSION} %{RELEASE} %{ARCH}\n",
+		)
+		if err != nil {
+			return fmt.Errorf("failed to list installed RPMs:\n%w", err)
+		}
+
+		lines := strings.Split(strings.TrimSpace(out), "\n")
+		for _, line := range lines {
+			parts := strings.Fields(line)
+			if len(parts) != 4 {
+				logger.Log.Infof("Skipping malformed RPM line while parsing installed RPMs for COSI: %q", line)
```

Comment:
not an expert on customizer's logging standards, but intuition tells me a warning or even error could be better candidates for this case